src/SmartComponents/DriftPage/SearchBar/SearchBar.js

```diff
@@ -10,13 +11,19 @@ export class SearchBar extends Component {
         super(props);
     }
 
+    updateFactFilter() {
+        const { changeFactFilter } = this.props;
+        let factFilter = document.getElementById('filterByFact').value;
+        changeFactFilter(factFilter);
+    }
+
     render() {
         return (
             <React.Fragment>
                 <TextInput
+                    id="filterByFact"
                     placeholder="Filter by Fact"
-                    value={ this.props.factFilter }
-                    onChange={ this.props.changeFactFilter }
+                    onChange={ _.debounce(this.props.changeFactFilter, 1000) }
```

Comment:
This amount of time, for me, made the application feel slow. I lowered it to 250 locally and things felt good while having a delay.